src/main/scala/org/allenai/ike/IkeBatchSearch.scala

```diff
@@ -0,0 +1,338 @@
+
+        case Failure(f) =>
+          logger.error(s"Unable to parse pattern  ${pattern.name}", f)
+      }
+    }
+
+    /** Split a pattern line into a pattern name and table name, as a tuple
+      *
+      * @param patternLine the pattern line to split
+      * @return a tuple with (pattern name, table name)
+      */
+    def splitPattern(patternLine: String): Option[(String, String)] = {
+      patternLine.split("\t").map(_.trim) match {
+        case Array(patternName, tableName, _*) => Some((patternName, tableName))
+        case _ => None
+      }
+    }
+
+    val sparkConf = new SparkConf()
+      .setAppName(this.getClass.getSimpleName)
+      .setMaster("local[*]")
+      .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .set("spark.kryo.registrator", "org.allenai.ike.IkeKryoRegistrator")
+    val sparkContext = new SparkContext(sparkConf)
+
+    // Open the index directory or create a temp directory if one hasn't been provided.
+    val indexParentDir = batchSearchOptions.indexPath.map { path =>
+      val dir = new File(path)
+      dir.mkdir()
+      dir
+    }.getOrElse {
+      val tempDir = Files.createTempDirectory("ike").toFile
+      tempDir.deleteOnExit()
+      tempDir
+    }
+
+    // Split the patterns and key them by table name. We also validate the pattern lines here.
+    val patternLines = sparkContext.textFile(batchSearchOptions.patternFile).collect
+    val patternsByTable = patternLines.flatMap { patternLine =>
+      // Process the pattern and run the query.
+      splitPattern(patternLine) match {
+        case Some(v) => Some(v)
+        case _ =>
+          logger.error("Invalid format in pattern file, line: " + patternLine)
+          None
+      }
+    }.groupBy(_._2).map {
+      case (tableName, splitLines) =>
+        if (splitLines.length == 1) {
+          (tableName, splitLines.head._1)
+        } else {
+          logger.error("Multiple patterns mapping to the same output table: " + tableName)
+          sys.exit(1)
+        }
+    }
+
+    // Find all existing index directories. If there are none, we'll need to build an index.
+    // TODO: Resuming a partial index would be a nice feature.
+    val indexDirectories = indexParentDir.listFiles().filter(_.isDirectory)
+    val indexPaths = if (indexDirectories.nonEmpty) {
+      indexDirectories.map(_.getAbsolutePath)
+    } else {
+      // Load the text source and partition it according to ideal index size. The maximum sentences
+      // per partition were determined through several experimental runs. As indices get larger,
+      // indexing slows down. Since partitioning in our scenario is arbitrary and relevance doesn't
+      // matter, we should optimize strictly for speed.
+      val textLines = sparkContext.textFile(batchSearchOptions.textSource)
+      val sentences = textLines.repartition {
+        val maxSentencesPerPartition = 256 * 1024
+        val minPartitions = Runtime.getRuntime.availableProcessors
```

Comment:
This is the number of processors on the driver? That's not useful when running on a cluster.